pkg/sessionctx/variable/session.go

```diff
@@ -3658,3 +3662,33 @@ func (s *SessionVars) PessimisticLockEligible() bool {
 	}
 	return false
 }
+
+// RemoveLockDDLJobs removes the DDL jobs which doesn't get the metadata lock from jobs.
+func RemoveLockDDLJobs(sv *SessionVars, jobs map[int64]*mdldef.JobMDL, printLog bool) {
+	if sv.InRestrictedSQL {
+		return
+	}
+	sv.TxnCtxMu.Lock()
+	defer sv.TxnCtxMu.Unlock()
+	if sv.TxnCtx == nil {
+		return
+	}
+	sv.GetRelatedTableForMDL().Range(func(tblID, value any) bool {
+		for jobID, jobMDL := range jobs {
+			if _, ok := jobMDL.TableIDs[tblID.(int64)]; ok && value.(int64) < jobMDL.Ver {
```

Comment:
Could you please add more comments about this logic? For example, does `jobs` represent all DDL jobs and their MDLs? Why we can delete this jobID if `value < jobMDL.Ver`? Thx~